### DISCUSSION TITLE
Set DHCP Hostname along with mDNS?

### DISCUSSION THREAD

**Discussion by SCarter858** [HomeSpan/HomeSpan#507] [Ideas]
Perhaps I'm missing something, but it seems that HomeSpan uses all its "set HostName" options only to set the name used by mDNS.  So when it starts WiFi and uses DHCP, it registers on the network using the default "esp32-xxx" hostname.  Kinda confusing to have 2 different Hostnames depending on how you look for the device on the network.  

Would it make sense to use the HomeSpan HostName when making the initial DHCP network registration?  I think this is as simple as a single call to WiFi.setHostname before starting WiFi.  I know I could add this manually, but since HomeSpan.begin() takes care of all network setup, it's cleaner to have it internal to that process...

**Comment by HomeSpan** [comment]
You are correct.  Recently I experimented with WiFi.setHostname and my router picks up the name correctly, but I did find an edge case that gave some pause.  Hostnames are generally unique, and by default HomeSpan will pick a unique name based on its device ID.  But these can be over-ridden and you can create the same name using two devices, which is not that uncommon if you are testing out the same sketch on an ESP32, ESP32-S2, etc (which I often do).   Some routers may have no problem handling two devices with the same name, but others may not, and I did not want to risk having HomeSpan cause someone's router to crash, or require a reboot.  This is not really a problem for MDNS since that's just point to point, and if you duplicate MDNS entries, it's easy to reverse (and any router that cached the info should refresh at some point).

It might make sense for me to make this an option so users can choose whether to over-ride the host name with the same name used for MDNS, or leave the default (i.e. esp32-XXX").

Thanks for raising this.

**Reply by andywaplinger** [reply]
@HomeSpan I'd also like to have that option. What I'd ultimately like to do is automatically create and change the hostname based on a device name I set as a variable.

For example, I'm creating an LED controller that I've named "Server Rack", which is one of several uniquely-named LED controllers. Before uploading the sketch, I manually change the `name` to `"Server Rack"`. Then I want to programmatically change the hostname to `"server-rack"` so I can access WebSerial just by going to `server-rack.local/webserial`.

If you're concerned about duplicate hostnames, is it reasonable to check the network for a matching hostname, and then automatically append a unique ID to the new device's hostname? And/or change the status LED to a different pattern indicating there's a duplicate hostname?

**Reply by HomeSpan** [reply]
To confirm, the option you would like is to be able to programmatically set the hostName to match the MDNS name from the sketch (e.g. something like homeSpan.matchHostName(boolean match))

**Reply by andywaplinger** [reply]
Yes, exactly. I just figured out how to automatically create and change the hostname based on the name, so now I'd like to also use that custom hostname as the mDNS hostname.

**Comment by tueddy** [comment]
I don't know if it's relevant here:

I have an ESPAsync webserver (Port 80) and mDNS already started in my project with hostname "myproject.local". With this URL I can also access all ESP32 websites.
After Homespan has been started (on Port 8080), the websites with this address are no longer accessible, only via IP address. How can I solve this?

**Reply by HomeSpan** [reply]
HomeSpan manages MDNS and sets a host name, which will override what you had set.  You can set the host name that HomeSpan uses for MDNS to anything you want (see the docs), such as "myproject", so no need to start MDNS separately.

**Reply by andywaplinger** [reply]
@tueddy here's my configuration, in case it's helpful https://github.com/andywaplinger/ESP32-HomeKit-LED

**Reply by tueddy** [reply]
`homeSpan.setHostNameSuffix("");` works fine for me, thank you!
